### PR TITLE
feat: add integration scheme migration script

### PR DESCRIPTION
**Pull request by djabarovgeorge**
### What change does this PR introduce?

integration migration script.

### Why was this change needed?

migration script in order to update the old document into the new scheme structure

### CODE REVIEW

File: apps/api/migrations/integration-scheme-update/add-integration-identifier-migration.ts
Comment (p-fernandez) [inline apps/api/migrations/integration-scheme-update/add-integration-identifier-migration.ts:87]:
```suggestion
  const environmentName = ENVIRONMENT_NAME_TO_SHORT_NAME[environment?.name] || '';
```
And in the object remove the entry for `undefined`. I think it can be simplified this logic like this.

Comment (djabarovgeorge) [reply apps/api/migrations/integration-scheme-update/add-integration-identifier-migration.ts:87]: OMG, I forgot to comment there apologies! I added it because of TS, but it could not happen in real life IMO added the empty string backoff as an unnecessary over-engineering guard.

File: apps/api/migrations/integration-scheme-update/add-integration-identifier-migration.ts
Comment (LetItRock) [inline apps/api/migrations/integration-scheme-update/add-integration-identifier-migration.ts:80]: This is how I'm going to generate these for the create endpoint, maybe you could do it similarly:
```
const providerIdCapitalized = `${command.providerId.charAt(0).toUpperCase()}${command.providerId.slice(1)}`;
const defaultName = providers.find((provider) => provider.id === command.providerId)?.displayName ?? providerIdCapitalized;
const name = command.name ?? defaultName;
const identifier = command.identifier ?? `${slugify(name, { lower: true, strict: true })}-${shortid.generate()}`;
```

Comment (p-fernandez) [reply apps/api/migrations/integration-scheme-update/add-integration-identifier-migration.ts:80]: It would be great if both places (migration and use case) share the same function, in my opinion. 👍🏻

File: apps/api/migrations/integration-scheme-update/add-integration-identifier-migration.spec.ts
Comment (djabarovgeorge) [reply apps/api/migrations/integration-scheme-update/add-integration-identifier-migration.spec.ts:38]: No reason, do you prefer another one here?

File: apps/api/migrations/integration-scheme-update/add-integration-identifier-migration.ts
Comment (djabarovgeorge) [inline apps/api/migrations/integration-scheme-update/add-integration-identifier-migration.ts:6]: Created the batched version so we could do it with less load on the DB.

File: apps/api/migrations/integration-scheme-update/add-integration-identifier-migration.spec.ts
Comment (p-fernandez) [reply apps/api/migrations/integration-scheme-update/add-integration-identifier-migration.spec.ts:38]: Sorry I didn't pay attention well and this is a test. No preference. All good. 😓

File: apps/api/migrations/integration-scheme-update/add-integration-identifier-migration.ts
Comment (djabarovgeorge) [reply apps/api/migrations/integration-scheme-update/add-integration-identifier-migration.ts:80]: > This is how I'm going to generate these for the create endpoint, maybe you could do it similarly:
> 
> ```
> const providerIdCapitalized = `${command.providerId.charAt(0).toUpperCase()}${command.providerId.slice(1)}`;
> const defaultName = providers.find((provider) => provider.id === command.providerId)?.displayName ?? providerIdCapitalized;
> const name = command.name ?? defaultName;
> const identifier = command.identifier ?? `${slugify(name, { lower: true, strict: true })}-${shortid.generate()}`;
> ```

Is this logic already merged with next? If not can you pin here the relevant PR id so I will know where it is, once it is merged i will reuse the same function logic here

Comment (LetItRock) [reply apps/api/migrations/integration-scheme-update/add-integration-identifier-migration.ts:80]: the PR: https://github.com/novuhq/novu/pull/3674

Comment (djabarovgeorge) [reply apps/api/migrations/integration-scheme-update/add-integration-identifier-migration.ts]: in order to not wait for the merge to next i committed duplication of this logic for now.

**Review by LetItRock — APPROVED** [review]

**Pull request merged**
